backend/internal/api/handlers.go

```diff
@@ -0,0 +1,299 @@
+// Package api provides HTTP handlers with repository access
+package api
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/ali01/mnemosyne/internal/config"
+	"github.com/ali01/mnemosyne/internal/models"
+	"github.com/ali01/mnemosyne/internal/repository"
+	"github.com/gin-gonic/gin"
+)
+
+// Handler holds dependencies for HTTP handlers
+type Handler struct {
+	repos *repository.Repositories
+	cfg   *config.Config
+}
+
+// getGraph returns the full graph data from the database
+func (h *Handler) getGraph(c *gin.Context) {
+	ctx := context.Background()
```

Comment:
Consider adding context timeout handling. While using context.Background() is fine for now, you might want to apply a timeout from the config or accept context from Gin for proper request cancellation support:
```go
ctx := c.Request.Context()
// or
ctx, cancel := context.WithTimeout(c.Request.Context(), 30*time.Second)
defer cancel()
```